MSAL/src/native_auth/controllers/sign_up/MSALNativeAuthSignUpController.swift

```diff
@@ -453,21 +519,35 @@ final class MSALNativeAuthSignUpController: MSALNativeAuthBaseController, MSALNa
                 self?.stopTelemetryEvent(event, context: context, delegateDispatcherResult: result)
             })
         case .error(let apiError):
-            let error = apiError.toVerifyCodePublicError()
+            let context = updateContext(context, with: apiError.getHeaderCorrelationId())
+            let error = apiError.toVerifyCodePublicError(context: context)
             stopTelemetryEvent(event, context: context, error: error)
             MSALLogger.log(level: .error,
                            context: context,
                            format: "Error in signup/continue request \(error.errorDescription ?? "No error description")")
             return .init(.error(error: error, newState: nil))
-        case .attributeValidationFailed:
-            let error = VerifyCodeError(type: .generalError)
+        case .attributeValidationFailed(let apiError, _):
+            let context = updateContext(context, with: apiError.getHeaderCorrelationId())
+            let error = VerifyCodeError(
```

Comment:
is there a `apiError.toVerifyCodeError()` available?

Reply:
Yes, we can use it here. I've merged the logic with the `case .error(let apiError)` 🙂